homeassistant/components/obihai/config_flow.py

```diff
@@ -43,21 +44,56 @@ class ObihaiFlowHandler(ConfigFlow, domain=DOMAIN):
 
     VERSION = 1
 
+    def __init__(self) -> None:
+        """Initialize."""
+        self._host: str | None = None
+
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Handle a flow initialized by the user."""
         errors: dict[str, str] = {}
 
         if user_input is not None:
-            self._async_abort_entries_match({CONF_HOST: user_input[CONF_HOST]})
+            await self.async_set_unique_id(user_input[CONF_HOST])
```

Comment:
Host is not valid as a unique id.
You have to keep async_abort_entries_match

Reply:
Discovery requires setting a unique ID, so I'm continuing down this road... Poking around the repo, I see it's not unusual to use serial, is that acceptable?  I just uploaded a draft with my thinking...